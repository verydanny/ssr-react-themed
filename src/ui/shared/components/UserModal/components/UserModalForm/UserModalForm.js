import React, { PureComponent } from 'react'
import { Form, Field } from '@rentpath/react-ui-core'
import PropTypes from 'prop-types'

class UserModalForm extends PureComponent {
  static propTypes = {
    fields: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string.isRequired,
        modal: PropTypes.string.isRequired,
      })
    ).isRequired,
    fieldMap: PropTypes.object,
    displayName: PropTypes.string.isRequired,
    onSubmit: PropTypes.func.isRequired,
  }

  static defaultProps = {
    fieldMap: {},
  }

  get getFields() {
    const { fields, fieldMap, displayName } = this.props

    return fields.map((field, i) => {
      const { name, modal } = field
      const Node = fieldMap[name] || Field

      return (
        <Node
          data-tid={`user-${modal}-form-${name}`}
          key={`${displayName}-${i}-${name}`}
          id={`${displayName}-${i}-${name}`}
          {...field}
        />
      )
    })
  }

  render() {
    const { onSubmit } = this.props
    return (
      <Form
        onSubmit={onSubmit}
      >
        {this.getFields}
      </Form>
    )
  }
}

export default UserModalForm
