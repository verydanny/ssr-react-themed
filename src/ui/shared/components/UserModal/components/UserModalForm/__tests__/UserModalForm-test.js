import React from 'react'
import { mount } from 'enzyme'
import { keyMirror } from '@rentpath/react-ui-utils'
import UserModalForm from '../UserModalForm'

const theme = keyMirror(['RegisterModal_FirstName'])

const fields = [
  {
    name: 'firstName',
    type: 'text',
    modal: 'register',
    placeholder: 'First Name',
    'data-tag_item': 'firstName',
    autoComplete: 'given-name',
    className: theme.RegisterModal_FirstName,
  },
]

const fieldMap = {}

const props = {
  theme,
  onSubmit: jest.fn(),
  displayName: 'RegisterModal',
  fields,
  fieldMap,
}

describe('UserModalForm', () => {
  describe('User modal form renders properly', () => {
    const wrapper = mount(<UserModalForm {...props} />)

    it('matches snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
