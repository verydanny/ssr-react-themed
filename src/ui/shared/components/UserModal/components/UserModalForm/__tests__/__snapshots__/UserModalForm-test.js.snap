// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserModalForm User modal form renders properly matches snapshot 1`] = `

<form action="#"
      method="POST"
      novalidate
      class
>
  <div class="RegisterModal_FirstName">
    <input id="RegisterModal-0-firstName"
           type="text"
           data-tid="user-register-form-firstName"
           name="firstName"
           modal="register"
           placeholder="First Name"
           data-tag_item="firstName"
           autocomplete="given-name"
           class
    >
  </div>
</form>

`;
